src/Storages/ExternalStream/Kafka/Kafka.cpp

```diff
@@ -166,4 +235,9 @@ void Kafka::validate()
 
     shards = result.partitions;
 }
+
+SinkToStoragePtr Kafka::write(const ASTPtr & /*query*/, const StorageMetadataPtr & metadata_snapshot, ContextPtr context)
+{
+    return std::make_shared<KafkaSink>(this, metadata_snapshot->getSampleBlock(), context, log);
```

Comment:
For multiple partitions, may need multiple `KafkaSink` likes `read method`.
We also need to consider how to partition